applications/openshift/kubelet/kubelet_configure_tls_min_version_master/rule.yml

```diff
@@ -0,0 +1,99 @@
+    custom profile one should be extremely careful as invalid configurations can be
+    catastrophic. It is always advised to configure minimum TLS version to TLSv1.2 or
+    latest when using Custom profile or to use predefined profiles Intermediate or
+    modern. If a TLS security profile is not configured, the default TLS security
+    profile is Intermediate.
+    </p>
+
+    <p>
+    To configure Custom tlsSecurityProfile for the Kubelet with TLSv1.2 as minimum
+    TLS version, create a new or modify existing <tt>KubeletConfig</tt> object along
+    these lines, one for every <tt>MachineConfigPool</tt>:
+      <pre>
+      apiVersion: machineconfiguration.openshift.io/v1
+      kind: KubeletConfig
+      metadata:
+          name: kubelet-tls-config-$pool
+      spec:
+          tlsSecurityProfile:
+              type: Custom
+              custom:
+                  ciphers:
+                  - ECDHE-ECDSA-CHACHA20-POLY1305
+                  - ECDHE-RSA-CHACHA20-POLY1305
+                  - ECDHE-RSA-AES128-GCM-SHA256
+                  - ECDHE-ECDSA-AES128-GCM-SHA256
+                  minTLSVersion: VersionTLS12
+          machineConfigPoolSelector:
+              matchLabels:
+                  pools.operator.machineconfiguration.openshift.io/$pool_name: ""
+      </pre>
+    </p>
+
+    <p>
+    For more information, follow
+    OpenShift documentation:
+    {{{ weblink(link="https://docs.openshift.com/container-platform/4.11/nodes/nodes/nodes-nodes-tls.html",
+                text="the relevant documentation") }}}.
+    </p>
+
+rationale: |-
+    The authenticity and integrity of the container platform and communication
+    between nodes and components must be secure. If an insecure protocol,
+    cipher, or algorithms is used, during transmission of data, the data can be
+    intercepted and manipulated. To thwart the manipulation of the data during
+    transmission secure protocol, cipher and algorithms must be used.
+
+severity: medium
+
+identifiers:
+    cce@ocp4: CCE-86723-4
+
+references:
+    nist: SC-8,SC-8(1)
+    srg: SRG-APP-000014-CTR-000040
+
+ocil_clause: 'Minimum TLS version configured for Kubelet uses insecure version'
+
+ocil: |-
+    Run the following command on the kubelet node(s):
+    <pre>$ for NODE_NAME in $(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz | jq '.kubeletconfig|.kind="KubeletConfiguration"|.apiVersion="kubelet.config.k8s.io/v1beta1" | .tlsMinVersion'; done</pre>
+    The output shows the configured TLS minimum version which should be
+    TLSv1.2 or newer.
+
+warnings:
+    - general: |-
+        {{{ openshift_cluster_setting_kubeletconfig() | indent(8) }}}
+
+template:
+    name: yamlfile_value
+    vars:
+        ocp_data: "true"
+        filepath: '/kubeletconfig/role'
+        filepath_suffix: var_role_master
+        yamlpath: ".tlsMinVersion"
+        values:
+            - value: "^(?!VersionTLS10|VersionTLS11)"
```

Comment:
same as above, we can use a variable to check against that variable

Reply:
comment incorporated.